server/service/src/document/encounter/update.rs

```diff
@@ -0,0 +1,262 @@
+use chrono::Utc;
+use repository::{Document, DocumentRepository, RepositoryError, TransactionError};
+
+use crate::{
+    document::{document_service::DocumentInsertError, raw_document::RawDocument},
+    service_provider::{ServiceContext, ServiceProvider},
+};
+
+use super::encounter_schema::SchemaEncounter;
+
+#[derive(PartialEq, Debug)]
+pub enum UpdateEncounterError {
+    InvalidParentId,
+    InvalidDataSchema(Vec<String>),
+    InternalError(String),
+    DatabaseError(RepositoryError),
+}
+
+pub struct UpdateEncounter {
+    pub parent: String,
+    pub data: serde_json::Value,
+    pub schema_id: String,
+}
+
+pub fn update_encounter(
+    ctx: &ServiceContext,
+    service_provider: &ServiceProvider,
+    store_id: String,
+    user_id: &str,
+    input: UpdateEncounter,
+) -> Result<Document, UpdateEncounterError> {
+    let patient = ctx
+        .connection
+        .transaction_sync(|_| {
+            let existing = validate(ctx, &input)?;
+            let doc = generate(user_id, input, existing)?;
+
+            // Updating the document will trigger an update in the patient (names) table
```

Comment:
Would encounter update patient (name) data ?

Reply:
No thanks this is a comment from the patient mutations.